src/scripts/main.js

```diff
@@ -1,3 +1,351 @@
+
+start.addEventListener('click', () => {
+  startGame();
+});
+
+window.addEventListener('keydown', (e) => {
+  if (e.code === 'ArrowRight' && isMoveRowsRight(arrayField)) {
+    arrayField.forEach(row => {
+      moveRight(row);
+    });
+
+    addNumbersIntoGameField(arrayField, gameOver);
+  }
+
+  if (e.code === 'ArrowLeft' && isMoveRowsLeft(arrayField)) {
+    arrayField.forEach(row => {
+      moveLeft(row);
+    });
+
+    addNumbersIntoGameField(arrayField, gameOver);
+  }
+
+  if (e.code === 'ArrowUp' && isMoveColumnsUp(arrayField)) {
+    moveUp(arrayField);
+    addNumbersIntoGameField(arrayField, gameOver);
+  }
+
+  if (e.code === 'ArrowDown' && isMoveColumnsDown(arrayField)) {
+    moveDown(arrayField);
+    addNumbersIntoGameField(arrayField, gameOver);
+  }
+
+  setGameField(arrayField, gameField);
+  isGameOver(arrayField, gameOver);
+  gameOver = deepClone(arrayField);
+});
+
+function deepClone(array) {
+  return JSON.parse(JSON.stringify(array));
+}
+
+function startGame() {
+  gameOver = [];
+  gameOver.length = arrayField.length;
+  gameOver.fill([]);
+  score = 0;
+  gameScore.innerText = 0;
+  messageLose.classList.add('hidden');
+  messageWin.classList.add('hidden');
+  messageRestart.classList.add('hidden');
+  messageWinner.classList.add('hidden');
+
+  clearGameField(arrayField);
+  addNumbersIntoGameField(arrayField, gameOver);
+  addNumbersIntoGameField(arrayField, gameOver);
+  setGameField(arrayField, gameField);
+}
+
+function randomizerCoords(field) {
+  const possibleY = [];
+
+  field.forEach((row, rowIndex) => {
+    for (let cellIndex = 0; cellIndex < row.length; cellIndex++) {
+      if (row[cellIndex] === 0) {
+        possibleY.push(rowIndex);
+        cellIndex = row.length;
+      }
+    }
+  });
+
+  if (possibleY.length === 0) {
+    return;
+  }
+
+  const y = possibleY[Math.floor(Math.random() * possibleY.length)];
+  const possibleX = [];
+
+  field[y].forEach((number, i) => {
+    if (!number) {
+      possibleX.push(i);
+    }
+  });
+
+  const x = possibleX[Math.floor(Math.random() * possibleX.length)];
+
+  return {
+    x,
+    y,
+  };
+}
+
+function randomizerNumber() {
+  return (Math.floor(Math.random() * 10) <= 8) ? 2 : 4;
+}
+
+function clearGameField(field) {
+  field.forEach((row, i) => {
+    row.forEach((cell, j) => {
+      field[i][j] = 0;
+    });
+  });
+}
+
+function setGameField(from, to) {
+  from.forEach((row, indexOfRow) => {
+    row.forEach((cell, indexOfCell) => {
+      let forCell = cell;
+
+      if (!forCell) {
+        forCell = '';
+      }
+
+      to.rows[indexOfRow].cells[indexOfCell].className = '';
+      to.rows[indexOfRow].cells[indexOfCell].classList.add('field-cell');
+
+      to.rows[indexOfRow].cells[indexOfCell].classList
+        .add(`field-cell--${forCell}`);
+
+      to.rows[indexOfRow].cells[indexOfCell].innerText = forCell;
+    });
+  });
+}
+
+function isMoveRowsRight(array) {
+  const isMoveAll = [];
+
+  array.forEach(row => {
```

Comment:
you can use the method `some` instead of `forEach` and return it without creating `isMoveAll`